### PR TITLE
RD-5776 Migrate Nodes widget to TypeScript

### PR DESCRIPTION
**Pull request by qooban**
## Description

Full migration of Nodes widget to TypeScript.

In the next PR (https://github.com/cloudify-cosmo/cloudify-stage/pull/2453) further refactoring in Nodes widget will be provided.

## Screenshots / Videos
N/A

## Checklist
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
1. [Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/4563/) - only Nodes widget test - failed due to external issues
2. Tested locally - PASSED.

## Documentation
N/A

### CODE REVIEW

File: widgets/nodes/src/NodesTable.tsx [hunk 1]
```diff
@@ -194,83 +209,3 @@ export default class NodesTable extends React.Component {
...
-                type_hierarchy: PropTypes.arrayOf(PropTypes.string)
-            })
-        ),
-        total: PropTypes.number
-    }).isRequired,
-    widget: Stage.PropTypes.Widget.isRequired,
-    toolbox: Stage.PropTypes.Toolbox.isRequired
-};
-
-function NodeTypeIcon({ typeHierarchy }) {
```
Comment (qooban) [inline widgets/nodes/src/NodesTable.tsx:225]: Moved to separate file.

File: widgets/nodes/src/NodesTable.tsx [hunk 2]
```diff
@@ -194,83 +209,3 @@ export default class NodesTable extends React.Component {
...
-    const icon = icons.getNodeIcon(_.reverse(_.clone(typeHierarchy)));
-
-    return <span style={{ fontSize: 20, fontFamily: 'cloudify' }}>{icon}</span>;
-}
-
-NodeTypeIcon.propTypes = {
-    typeHierarchy: PropTypes.arrayOf(PropTypes.string).isRequired
-};
-
-function TypeHierarchyTree({ typeHierarchy }) {
```
Comment (qooban) [inline widgets/nodes/src/NodesTable.tsx:235]: Moved to separate file.

File: app/widgets/common/deploymentsView/types.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ export enum DeploymentStatus {
     RequiresAttention = 'requires_attention'
 }
 
-export interface Deployment {
+export interface FullDeployment {
```
Comment (qooban) [inline app/widgets/common/deploymentsView/types.ts:21]: It's not yet full, but ready to be extended as part of different PRs.

File: widgets/nodes/src/widget.tsx [hunk 4]
```diff
@@ -1,15 +1,47 @@
...
+    const groups: Record<string, string[]> = {};
+    _.forEach(deployments, deployment => {
+        _.forIn(deployment.groups, (group, groupId) => {
+            _.forEach(group.members, nodeId => {
+                groups[nodeId + deployment.id] = groups[nodeId + deployment.id] || [];
+                const groupList = groups[nodeId + deployment.id];
+                groupList.push(groupId);
+            });
+        });
+    });
```
Comment (kubama) [inline widgets/nodes/src/widget.tsx:19]: We could try to move to vanilla JS once we have typing in place

**Review by kubama — APPROVED** [review]
Looks very good, not much to comment on :)

File: widgets/nodes/src/widget.tsx (hunk 4 above)
Comment (qooban) [reply widgets/nodes/src/widget.tsx:19]: Sure, I'll do that in https://github.com/cloudify-cosmo/cloudify-stage/pull/2453.

File: app/widgets/common/deploymentsView/types.ts (hunk 3 above)
Comment (qooban) [reply app/widgets/common/deploymentsView/types.ts:21]: @Blewin FYI - this change will introduce a conflict in #2426 , but I suspect it'll be quick to fix.

Comment (qooban) [reply app/widgets/common/deploymentsView/types.ts:21]: After our today's discussion created https://github.com/cloudify-cosmo/cloudify-stage/pull/2457 so it should be a bit more clear with it.
